### DISCUSSION TITLE
How hide next Action in wizard?

### DISCUSSION THREAD

**Discussion by sh4msi** [filamentphp/filament#9942] [Help]
How can I hide the next button in the wizard?
I tested with the following code, but it didn't work.

```
            Wizard::make()
                ->schema([
                    $this->getStepOne(),
                    $this->getStepTwo(),
                    $this->getStepThree(),
                ])
                ->nextAction(
                    fn (Action $action) => $action->hidden()
                )
```

**Comment by leandrocfe** [comment]
try this
```php
->nextAction(fn (Action $action): Action => $action->extraAttributes(['x-show' => 'false']))
```

**Reply by sh4msi** [reply]
Thank you for your time and answer

Mostly my goal was to see if there is such a problem or not
Because it's not just hidden/visible
For example, when we disable it, it appears to be disabled, but the action still works.

------
I managed to hide it in this way for now

```
                ->nextAction(
                    fn(Action $action) =>  Action::make('test')
                )
```

**Reply by leandrocfe** [reply]
Worked?

**Reply by sh4msi** [reply]
yes it worked.
But this is not the answer I want.

**Reply by leandrocfe** [reply]
Please explain what you need and share the whole code you are trying to do

**Reply by sh4msi** [reply]
I want to know why this code does not work

`fn (Action $action) => $action->hidden()`
or this
`fn (Action $action) => $action->disabled()`

**Reply by leandrocfe** [reply]
I think it won't work because the click action is in the span
```php
<span
    x-cloak
    x-on:click="
        $wire.dispatchFormEvent(
            'wizard::nextStep',
            '{{ $statePath }}',
            getStepIndex(step),
        )
    "
    x-show="! isLastStep()"
>
    {{ $getAction('next') }}
</span>
```
take a look https://github.com/filamentphp/filament/blob/3.x/packages/forms/resources/views/components/wizard.blade.php

**Reply by sh4msi** [reply]
Are you sure about this?
I tested and changed to div, but I still didn't get the answer.

**Comment by sh4msi** [comment]
I think this is a bug
what is your opinion?

@danharrin 
@leandrocfe

**Comment by leandrocfe** [comment]
> I think this is a bug
> what is your opinion?
> 
> @danharrin 
> @leandrocfe

Open an issue 🤷‍♂️

**Reply by danharrin** [reply]
with a reproduction repository
